src/components/group-common.js

```diff
@@ -104,76 +105,115 @@ export const NewUserModal = ajaxCaller(class NewUserModal extends Component {
   }
 
   render() {
-    const { adminLabel, userLabel, title, onDismiss, footer } = this.props
-    const { userEmail, roles, suggestions, submitting, submitError } = this.state
-
-    const errors = validate({ userEmail }, { userEmail: { email: true } })
+    const { adminLabel, userLabel, title, onDismiss, footer, ajax: { User } } = this.props
+    const { confirmAddUser, userEmail, roles, suggestions, busy, submitError } = this.state
+    const emailConstraints = { format: /\w+@(gmail\.com|broadinstitute\.org)$/ }
+    const errors = validate({ userEmail }, { userEmail: emailConstraints })
     const isAdmin = _.includes(adminLabel, roles)
 
-    const userEmailInvalid = !!validate({ userEmail }, { userEmail: { email: true } })
+    const userEmailInvalid = !!validate({ userEmail }, { userEmail: emailConstraints })
     const canAdd = value => value !== userEmail || !userEmailInvalid
 
-    return h(Modal, {
-      onDismiss,
-      title,
-      okButton: h(ButtonPrimary, {
-        tooltip: Utils.summarizeErrors(errors),
-        onClick: () => this.submit(),
-        disabled: errors
-      }, ['Add User'])
-    }, [
-      h(RequiredFormLabel, ['User email']),
-      h(AutocompleteSearch, {
-        autoFocus: true,
-        value: userEmail,
-        onChange: v => this.setState({ userEmail: v }),
-        renderSuggestion: suggestion => div({ style: styles.suggestionContainer }, [
-          div({ style: { flex: 1 } }, [
-            !canAdd(suggestion) && h(TooltipTrigger, {
-              content: 'Not a valid email address'
-            }, [
-              icon('warning-standard', { style: { color: colors.danger(), marginRight: '0.5rem' } })
-            ]),
-            suggestion
-          ])
-        ]),
-        onSuggestionSelected: selection => {
-          this.setState({ userEmail: selection })
-        },
-        onKeyDown: e => {
-          // 27 = Escape
-          if (e.which === 27 && !!userEmail) {
-            this.setState({ userEmail: '' })
-            e.stopPropagation()
+    return Utils.cond(
+      [confirmAddUser, () => h(Modal, {
+        title: 'User is not registered',
+        okButton: h(ButtonPrimary, {
+          onClick: async () => {
+            this.setState({ busy: true })
```

Comment:
We should generally use `withBusyState` which helps ensure there isn't an infinite spinner in case of an error.